CollectionViewShelfLayout/CollectionViewShelfLayout.swift

```diff
@@ -180,6 +184,13 @@ open class CollectionViewShelfLayout: UICollectionViewLayout {
       }
     }
   }
+    
+  open override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+    guard let collectionView = collectionView else {
+      return true
+    }
+    return collectionView.frame.size != newBounds.size
```

Comment:
Since we're supporting only the vertical layout. I think it would be better to compare just the width